### PR TITLE
Dispatcher updates in support of PeerManager integration with splinterd/AdminService

### PR DESCRIPTION
**Pull request by agunde406**

### CODE REVIEW

File: libsplinter/src/network/dispatch.rs [hunk 1]
```diff
@@ -602,6 +618,37 @@ where
     }
 }
 
+impl<MT, Source> DispatchLoop<MT, Source>
+where
+    MT: Any + Hash + Eq + Debug + Clone + Send,
+    Source: Send + 'static,
+{
+    pub fn builder() -> DispatchLoopBuilder<MT, Source> {
```
Comment (peterschwarz) [inline libsplinter/src/network/dispatch.rs:610]: This should probably be in a separate commit

Comment (agunde406) [reply libsplinter/src/network/dispatch.rs:610]: split out

Comment (agunde406) [reply libsplinter/src/network/dispatch.rs:610]: split out

File: libsplinter/src/network/dispatch.rs [hunk 2]
```diff
@@ -289,6 +289,8 @@ pub trait MessageSender<R>: Send {
     ///
     /// If an error occurs, return the intended recipient and message bytes.
     fn send(&self, reciptient: R, message: Vec<u8>) -> Result<(), (R, Vec<u8>)>;
+
+    fn box_clone(&self) -> Box<dyn MessageSender<R>>;
```
Comment (peterschwarz) [inline libsplinter/src/network/dispatch.rs:293]: Why is this necessary?

Comment (agunde406) [reply libsplinter/src/network/dispatch.rs:293]: I added it for passing it into the create_authorization_dispatcher and thought it would be useful but I will rework that and remove it.

Comment (agunde406) [reply libsplinter/src/network/dispatch.rs:293]: auth2::handlers::create_authorization_dispatcher takes a MessageSender<ConnectionId>> and that needs to be wrapped in Box. That function clones the message sender to create two dispatchers.

Comment (agunde406) [reply libsplinter/src/network/dispatch.rs]: I was able to rework that function. removed box_clone
